SingularityService/src/main/java/com/hubspot/singularity/data/history/SingularityTaskHistoryPersister.java

```diff
@@ -88,21 +88,40 @@ public void runActionOnPoll() {
           int forRequest = 0;
           int transferred = 0;
           for (SingularityTaskId taskId : taskIds) {
-            if (moveToHistoryOrCheckForPurge(taskId, forRequest)) {
-              LOG.debug("Transferred task {}", taskId);
-              transferred++;
+            if (
+              configuration.skipPersistingTooLongTaskIds() &&
+              taskId.getId().length() > 200
+            ) {
+              if (
+                System.currentTimeMillis() -
+                taskId.getCreateTimestampForCalculatingHistoryAge() >
+                TimeUnit.DAYS.toMillis(7)
+              ) {
+                LOG.warn(
+                  "Deleting {} from ZK, could not persist in DB because of task ID length",
+                  taskId.getId()
+                );
+                purgeFromZk(taskId);
+              }
+              LOG.error("Task ID {} too long to pesist to DB, skipping", taskId.getId());
```

Comment:
nit, can put that log line in an else so we don't get the double log line when it's actually deleted and not skipped

Reply:
oh whoops, yes